### PR TITLE
Fix incorrect length in File.WriteAllTextAsync

### PR DESCRIPTION
**Pull request by stephentoub**
Fixes https://github.com/dotnet/corefx/issues/20443
cc: @danmosemsft, @JeremyKuhne, @ianhays

### CODE REVIEW

File: src/System.IO.FileSystem/tests/File/ReadWriteAllTextAsync.cs [hunk 1]
```diff
@@ -61,10 +62,13 @@ public class File_ReadWriteAllTextAsync : FileSystemTest
         [Theory]
         [InlineData(0)]
         [InlineData(100)]
+        [InlineData(4096)]
```
Comment (danmosemsft) [inline src/System.IO.FileSystem/tests/File/ReadWriteAllTextAsync.cs]: is 4095 intersting?

File: src/System.IO.FileSystem/src/System/IO/File.cs [hunk 2]
```diff
@@ -1005,7 +1005,7 @@ public static Task WriteAllLinesAsync(string path, IEnumerable<string> contents,
                 int index = 0;
                 while (index < count)
```
Comment (danmosemsft) [inline src/System.IO.FileSystem/src/System/IO/File.cs]: consider renaming `count` to `length`? Seems easier to read. `count` is typically a variable amount.

File: src/System.IO.FileSystem/tests/File/ReadWriteAllTextAsync.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.IO.FileSystem/tests/File/ReadWriteAllTextAsync.cs]: 100 seemed to cover it as a value less than the built-in buffer size, combined with changing the input to actually verify the right sequence of values.

File: src/System.IO.FileSystem/src/System/IO/File.cs (hunk 2 above)
Comment (stephentoub) [inline src/System.IO.FileSystem/src/System/IO/File.cs]: Could be done separately.  I simply want to fix the bug here with the most concise fix possible.

Comment (stephentoub) [inline src/System.IO.FileSystem/src/System/IO/File.cs]: Also, `count` is typically used to mean "number of items being copied", e.g. when taking buffer/offset/count triplets.

**Pull request merged**